core/src/main/scala/org/genivi/sota/core/storage/S3PackageStore.scala

```diff
@@ -42,28 +42,32 @@ class S3PackageStore(credentials: S3Credentials)
     client
   }
 
-  override def store(packageId: PackageId,
-                     filename: String, fileData: Source[ByteString, Any]): Future[(Uri, PackageSize, DigestResult)] = {
-    val tempFile = File.createTempFile(filename, ".tmp")
-
+  def storeAndRemove(packageId: PackageId,
+                     file: File, fileData: Source[ByteString, Any]): Future[(Uri, PackageSize, DigestResult)] = {
     // The s3 sdk requires us to specify the file size if using a stream
     // so we always need to cache the file into the filesystem before uploading
-    val sink = FileIO.toPath(tempFile.toPath)
+    val sink = FileIO.toPath(file.toPath)
       .mapMaterializedValue {
         _.flatMap { result =>
           if(result.wasSuccessful) {
-            upload(tempFile, filename, fileData)
-              .andThen { case _ =>
-                Try(tempFile.delete())
-              }
-          } else
+            upload(file, file.getName, fileData)
+          } else {
             Future.failed(result.getError)
+          }
         }
       }
 
     writePackage(packageId, fileData, sink)
+      .andThen { case _ =>
+        log.debug(s"removing temp file ${file.getAbsolutePath}")
+        Try(file.delete())
+      }
   }
 
+  override def store(packageId: PackageId, filename: String, fileData: Source[ByteString, Any]):
+                                                                          Future[(Uri, PackageSize, DigestResult)] =
+    storeAndRemove(packageId, File.createTempFile(filename, ".tmp"), fileData)
```

Comment:
Is `storeAndRemove` used somewhere else or can we "squash" `store` and `storeAndRemove` into single function?